app/models/parent_object.rb

```diff
@@ -42,19 +42,27 @@ def finished_states
     ['solr-indexed', 'pdf-generated']
   end
 
-  def create_child_records(_current_batch_process, _current_batch_connection = current_batch_connection)
+  def create_child_records
+    return unless ladybird_json
+    return self.child_object_count = 0 if ladybird_json["children"].empty?
+    # Note - the upsert_all method skips ActiveRecord callbacks, and is entirely
+    # database driven. This also makes object creation much faster.
+    ChildObject.upsert_all(array_of_child_hashes)
+    self.child_object_count = child_objects.size
```

Comment:
The counter is moved here - just a git diff quirk. We're indexing it to Solr in order to display how many images are available for a parent object in Blacklight.